panic: runtime error: slice bounds out of range [-1:]

When running pings in coroutines after a few "threads" finish i get this error/panic: 

```
panic: runtime error: slice bounds out of range [-3:]

goroutine 183 [running]:
github.com/dreamscached/minequery/v2.(*Pinger).ping17ReadStatusResponsePacketPayload(0xc0000564a0?, {0x2673ea66f98, 0xc0065da1c0})
        C:/Users/Sandbox/go/pkg/mod/github.com/dreamscached/minequery/v2@v2.3.0/ping_17.go:1841 +0x525
```

This is my code:
```go
var wg sync.WaitGroup
var pingout []minequery.Status17
for i, data := range payload {
  wg.Add(1)

  go func(data Data, i int) {
    defer wg.Done()

    res, err := minequery.Ping17(data.IP, data.Ports[0].Port)
    if err != nil { fmt.Printf("[%d -> %s] (Failed) %s\n", i, data.IP, err); return }
    pingout = append(pingout, *res)
  }(data, i)
}
wg.Wait()
```

Actually, managed to reproduce, it seems like a DDoS measure of some sort is used.

```
Testuj za darmo przez 24h! <nil> false false}
&{??? 0 0 1 [] Craftserve.pl - wydajny hosting Minecraft!
Testuj za darmo przez 24h! <nil> false false}
&{??? 0 0 1 [] Ochrona DDoS: Przekroczono limit polaczen. (1) <nil> false false}
&{??? 0 0 1 [] Ochrona DDoS: Przekroczono limit polaczen. (1) <nil> false false}
&{??? 0 0 1 [] Ochrona DDoS: Przekroczono limit polaczen. (1) <nil> false false}
panic: runtime error: slice bounds out of range [-1:]

goroutine 1 [running]:
github.com/dreamscached/minequery/v2.(*Pinger).ping17ReadStatusResponsePacketPayload(0xc00006c100?, {0x598320, 0xc000014608})
        /home/dreamscached/Dokumenty/Vývoj/Projekty/dreamscached/minequery/ping_17.go:1841 +0x525
github.com/dreamscached/minequery/v2.(*Pinger).Ping17(0xc00006c100, {0x565026, 0xe}, 0x1?)
        /home/dreamscached/Dokumenty/Vývoj/Projekty/dreamscached/minequery/ping_17.go:1757 +0x27b
github.com/dreamscached/minequery/v2.Ping17(...)
        /home/dreamscached/Dokumenty/Vývoj/Projekty/dreamscached/minequery/ping_17.go:1731
main.main()
        /home/dreamscached/Dokumenty/Vývoj/Projekty/dreamscached/minequery/main/main.go:11 +0x8b

Process finished with the exit code 2
```

--- a/README.md
+++ b/README.md
@@ -31,16 +31,15 @@ See [MIGRATING.md][1] for help with migrating from MineQuery.
 
 ## #️⃣ Minecraft Version Support
 
-As of version 2.0.0, MineQuery supports pinging of all versions of Minecraft.
+MineQuery supports pinging of all versions of Minecraft.
 
-| [Beta 1.8 to 1.3][2] | [1.4][3]    | [1.6 to 1.7][4] | [1.7+][5]   |
-|----------------------|-------------|-----------------|-------------|
-| ✅ Supported          | ✅ Supported | ✅ Supported     | ✅ Supported |
+| [Beta 1.8 to 1.3][2] | [1.4 to 1.5][3] | [1.6][4]    | [1.7+][5]   |
+|----------------------|-----------------|-------------|-------------|
+| ✅ Supported          | ✅ Supported     | ✅ Supported | ✅ Supported |
 
 ### Query Protocol Support
 
-As of version 2.0.0, query protocol is not yet supported.
-See [issue #25][6] to track progress.
+MineQuery fully supports [Query][9] protocol. 
 
 ## 📚 How to use
 
@@ -123,3 +122,4 @@ a convenient set of constants you can use &mdash; see `Ping17ProtocolVersion*` c
 [6]: https://github.com/alteamc/minequery/issues/25
 [7]: https://pkg.go.dev/github.com/alteamc/minequery/v2
 [8]: #basic-usage
+[9]: https://wiki.vg/Query

--- a/README.md
+++ b/README.md
@@ -15,7 +15,7 @@
     </a>
     <br/>
     <a href="https://github.com/alteamc/minequery#readme">
-        <img alt="Minecraft version support badge" src="https://img.shields.io/badge/minecraft%20version-Beta%201.8%20to%201.3%20%7C%201.4%20%7C%201.5%20to%201.6%20%7C%201.7%2B-brightgreen">
+        <img alt="Minecraft version support badge" src="https://img.shields.io/badge/minecraft%20version-Beta%201.8%20to%201.3%20%7C%201.4%20to%201.5%20%7C%201.6%20%7C%201.7%2B-brightgreen">
     </a>
     <a href="https://discord.gg/9ruheUG3Wg">
         <img alt="Discord server badge" src="https://discordapp.com/api/guilds/929337829610369095/widget.png?style=shield">

--- a/query.go
+++ b/query.go
@@ -64,6 +64,7 @@ type FullQueryStatus struct {
 	SamplePlayers []string
 	Port          int
 	Host          string
+	Data          map[string]string
 }
 
 // QueryBasic queries Minecraft servers and returns simplified query response.
@@ -510,6 +511,7 @@ func (p *Pinger) parseQueryFullStatResponse(reader io.Reader) (*FullQueryStatus,
 		SamplePlayers: players,
 		Port:          int(port),
 		Host:          hostname,
+		Data:          fields,
 	}, nil
 }
 
@@ -549,7 +551,7 @@ func queryParseFullStatPluginsList(str string) (string, []FullQueryPluginEntry, 
 	// Split version string by color; left part is server version and brand, right part is plugins list
 	parts := strings.SplitN(str, ":", 2)
 	if len(parts) < 2 {
-		return parts[0], nil, nil
+		return parts[0], make([]FullQueryPluginEntry, 0), nil
 	}
 	ver, rem := parts[0], parts[1]
 
@@ -574,6 +576,7 @@ func queryGetFullStatField(fields map[string]string, key string) (string, error)
 	if !ok {
 		return "", fmt.Errorf("%w: response body does not contain %s field", ErrInvalidStatus, key)
 	}
+	delete(fields, key)
 	return value, nil
 }
 

